File: php7/src/Conjured.php
<?php


namespace App;


class Conjured
{
    private $item;

    public function __construct($item) {
        $this->item = $item;
    }

    public function updateQuality(){
        $this->item->quality -= 2;

        if ($this->item->sell_in <= 0) {
            $this->item->quality -= 2;
        }

        $this->item->sell_in -= 1;
    }
}

File: Java/src/main/java/com/gildedrose/model/Conjured.java
package com.gildedrose.model;

import static com.gildedrose.Constants.MIN_QUALITY;

public class Conjured extends Goods {

    public Conjured(int sellIn, int quality) {
        super("Conjured Mana Cake", sellIn, quality);
    }

    @Override
    public void updateQuality() {
        if (sellIn > 0) {
            degradeQualityBy(2);
        } else {
            degradeQualityBy(4);
        }
        sellInPasses();
    }

    private void degradeQualityBy(int degradeBy) {
        quality = Math.max(quality - degradeBy, MIN_QUALITY);
    }
}
